### PR TITLE
Make `unique_together` work more seamlessly.

### PR DESCRIPTION
**Pull request by maxpeterson**
For #7

Add Orderable.validate_unique to exclude `sort_order` when it is unique_together with something.

### CODE REVIEW

**Comment by maxpeterson** [comment]
@meshy please review

File: orderable/models.py [hunk 1]
```diff
@@ -52,6 +52,23 @@ def prev(self):
 
         return self.get_filtered_manager().before(self)
 
+    def validate_unique(self, exclude=None):
+        if self._is_sort_order_unique_together_with_something():
+            exclude = exclude or []
+            if 'sort_order' not in exclude:
+                exclude.append('sort_order')
+        return super().validate_unique(exclude=exclude)
```
Comment (meshy) [inline orderable/models.py]: We're still supporting python 2 for the moment, so:

```suggestion
        return super(Orderable, self).validate_unique(exclude=exclude)
```

**Comment by meshy** [comment]
It's not entirely fair for the commit message to say "fixes #7"... To truly fix it we need a way to ensure that the `sort_order` field is either `unique=True`, or in `unique_together`.

File: orderable/models.py (hunk 1 above)
Comment (maxpeterson) [reply orderable/models.py]: Fail - I am so used to working only on Python 3 project code!

**Comment by maxpeterson** [comment]
@meshy I interpreted #7 as the tile

>`unique_together` doesn't work seamlessly

The issue does include a discussion about how to ensure that the `sort_order` field is unique or unique_together, but I saw that is a different issue. Either way we can either keep #7 open or start a new issue (which might have the benefit of not making me feeling so old)

**Pull request merged**